File: longestPalindrome.py
#Leetcode Problem : 409
#first add to set if already present in set remove from set and add 2 to count. At the end if there are elements
#in the set then return count+1 else return count

class Solution:
    def longestPalindrome(self, s):
        count=0
        c=set()
        for i in s:
            if i not in c:
                c.add(i)
            else:
                count+=2
                c.remove(i)
        if c:
            return count+1
        else:
            return count

#TC: O(n)
#SC:O(1)

File: LongestPalindrome.java
//TC: O(N) travelling once with HashSet
//SC: O(1) as only 26 lowercase alphabets will be present
//Running: yes

class Solution {
    public int longestPalindrome(String s) {
        HashSet<Character> set=new HashSet<>();
        int count=0;
        for(int i=0;i<s.length();i++){
            char curr= s.charAt(i);
            //Adding 2 elements to the count and removing the element from the hashSet;
            
            if(set.contains(curr)){
                set.remove(curr);
                count=count+2;
            }else{
                set.add(curr);
            }
        }
        if(!set.isEmpty()){
            count=count+1;
        }
        return count;
    }
}
